projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid-navigation.service.ts

```diff
@@ -179,6 +179,47 @@ export class IgxHierarchicalGridNavigationService extends IgxGridNavigationServi
         }
     }
 
+    /**
+     * Navigates to the specific child grid based on the array of paths leading to it
+     *
+     * @param pathToChildGrid Array of IPathSegments that describe the path to the child grid
+     * each segment is described by the rowKey of the parent row and the rowIslandKey.
+     */
+    public navigateToChildGrid(pathToChildGrid: IPathSegment[], cb?: () => void) {
+        if (pathToChildGrid.length == 0) {
+            if (cb) {
+                cb();
+            }
+            return;
+        }
+        const pathElem = pathToChildGrid.shift();
+        const rowKey = pathElem.rowKey;
+        const rowIndex = this.grid.gridAPI.get_rec_index_by_id(rowKey, this.grid.dataView);
```

Comment:
Probably a good idea to check whether `rowIndex` is -1 (<0) because it is causing a pending navigation.
A case where it is reproducible is:
In a hgrid, try to use the API with a key that does not exist in the parent, i.e (there is not a rowKey = 16) 
`this.grid.navigation.navigateToChildGrid([{rowID:"16",rowKey: "16", rowIslandKey: "childData"}]);`
Nothing happens.
But if you try to expand a row then an error is thrown:
![Screenshot 2025-01-06 at 18 09 51](https://github.com/user-attachments/assets/b69f7045-872a-410d-bb41-b3c4e3aab82f)

Reply:
Checks are added.